diff --git a/code/AssetLib/MS3D/MS3DLoader.cpp b/code/AssetLib/MS3D/MS3DLoader.cpp
index 703abcb868..213f3627e6 100644
--- a/code/AssetLib/MS3D/MS3DLoader.cpp
+++ b/code/AssetLib/MS3D/MS3DLoader.cpp
@@ -630,8 +630,7 @@ void MS3DImporter::InternReadFile( const std::string& pFile,
             if ((*it).posFrames.size()) {
                 nd->mPositionKeys = new aiVectorKey[(*it).posFrames.size()];
 
-                aiQuatKey* qu = nd->mRotationKeys;
-                for(std::vector<TempKeyFrame>::const_iterator pos = (*it).posFrames.begin(); pos != (*it).posFrames.end(); ++pos,++qu) {
+                for(std::vector<TempKeyFrame>::const_iterator pos = (*it).posFrames.begin(); pos != (*it).posFrames.end(); ++pos) {
                     aiVectorKey& v = nd->mPositionKeys[nd->mNumPositionKeys++];
 
                     v.mTime = (*pos).time*animfps;
diff --git a/contrib/stb/stb_image.h b/contrib/stb/stb_image.h
index 9eedabedc4..2db4cca0a9 100644
--- a/contrib/stb/stb_image.h
+++ b/contrib/stb/stb_image.h
@@ -5079,7 +5079,7 @@ static int stbi__parse_png_file(stbi__png *z, int scan, int req_comp)
 {
    stbi_uc palette[1024], pal_img_n=0;
    stbi_uc has_trans=0, tc[3]={0};
-   stbi__uint16 tc16[3];
+   stbi__uint16 tc16[3] = {0};
    stbi__uint32 ioff=0, idata_limit=0, i, pal_len=0;
    int first=1,k,interlace=0, color=0, is_iphone=0;
    stbi__context *s = z->s;
diff --git a/code/AssetLib/MS3D/MS3DLoader.cpp b/code/AssetLib/MS3D/MS3DLoader.cpp
index 213f3627e6..5eff73348b 100644
--- a/code/AssetLib/MS3D/MS3DLoader.cpp
+++ b/code/AssetLib/MS3D/MS3DLoader.cpp
@@ -133,7 +133,8 @@ void MS3DImporter :: ReadComments(StreamReaderLE& stream, std::vector<T>& outp)
 }
 
 // ------------------------------------------------------------------------------------------------
-template <typename T, typename T2, typename T3> bool inrange(const T& in, const T2& lower, const T3& higher)
+template <typename T, typename T2, typename T3> 
+bool inrange(const T& in, const T2& lower, const T3& higher)
 {
     return in > lower && in <= higher;
 }
@@ -608,7 +609,7 @@ void MS3DImporter::InternReadFile( const std::string& pFile,
         // anim->mDuration = totalframes/animfps;
 
         anim->mChannels = new aiNodeAnim*[joints.size()]();
-        for(std::vector<TempJoint>::const_iterator it = joints.begin(); it != joints.end(); ++it) {
+        for (auto it = joints.begin(); it != joints.end(); ++it) {
             if ((*it).rotFrames.empty() && (*it).posFrames.empty()) {
                 continue;
             }
@@ -618,7 +619,7 @@ void MS3DImporter::InternReadFile( const std::string& pFile,
 
             if ((*it).rotFrames.size()) {
                 nd->mRotationKeys = new aiQuatKey[(*it).rotFrames.size()];
-                for(std::vector<TempKeyFrame>::const_iterator rot = (*it).rotFrames.begin(); rot != (*it).rotFrames.end(); ++rot) {
+                for (auto rot = (*it).rotFrames.begin(); rot != (*it).rotFrames.end(); ++rot) {
                     aiQuatKey& q = nd->mRotationKeys[nd->mNumRotationKeys++];
 
                     q.mTime = (*rot).time*animfps;
@@ -630,7 +631,7 @@ void MS3DImporter::InternReadFile( const std::string& pFile,
             if ((*it).posFrames.size()) {
                 nd->mPositionKeys = new aiVectorKey[(*it).posFrames.size()];
 
-                for(std::vector<TempKeyFrame>::const_iterator pos = (*it).posFrames.begin(); pos != (*it).posFrames.end(); ++pos) {
+                for(auto pos = (*it).posFrames.begin(); pos != (*it).posFrames.end(); ++pos) {
                     aiVectorKey& v = nd->mPositionKeys[nd->mNumPositionKeys++];
 
                     v.mTime = (*pos).time*animfps;
